v2/pkg/engine/postprocess/postprocess.go

```diff
@@ -161,3 +177,16 @@ func (p *Processor) createFetchTree(res *resolve.GraphQLResponse) {
 		ChildNodes: children,
 	}
 }
+
+// collectDataSourceInfos returns the list of involved data sources of the operation
+func collectDataSourceInfos(node *resolve.FetchTreeNode) (list []resolve.DataSourceInfo) {
```

Comment:
btw, you could do this in the createFetchTree

there is a step when you got all fetches slice
ao you could just iterate and collect datasource in one pass without the need to traverse the tree

Reply:
Great suggestion. I'll change it.